### DISCUSSION TITLE
When Copilot works, it's awesome. But sometimes it's really frustrating. (An airing of polite grievances.)

### DISCUSSION THREAD

**Discussion by alexepperly** [community/community#46526] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

See title.  This might be a controversial take, but here goes:  When GitHub Copilot works, it's fantastic. Scary, even.  I've had it complete some incredibly complex code for me, and also take the boring repetition out of stuff that I would have had to write manually before.

But there are a mounting list of issues that are hard to ignore. 

1. **The Big One: Sometimes Copilot just decides that it's not going to give me suggestions, even for very, VERY, simple operations.** Earlier today, I was trying to make a simple barplot in pandas, and no matter what I did, it would not suggest anything to me, no matter how detailed my comments were.  Don't get me wrong, I can write this code myself pretty easily, but this happens pretty frequently, and is the exact type of thing that Copilot is supposed to save us time with.  I have looked through the issues here and in other places, and I see suggestions from Microsoft internal folks of how to restructure your code on the page to get these suggestions to work.  But frankly -- and respectfully -- if we're paying money for this service, these kinds of bugs/quirks should be worked out, and/or there should be clear warnings/guides within the extension itself about these issues.  Having to find the answers buried in other message boards doesn't seem to be the way to go. 

2. **There should be an easy, intuitive UI to interact with Copilot, and one that is easy and intuitive to access.** Copilot doesn't really have a UI, unless you count the little button at the bottom of VS Code.  I know we big strong devs are way too smart and sophisticated for a UI, but just admit it -- it would be nice to have more visual representation of what is going on.  Yes, you can hit Ctrl-Enter to open Copilot suggestions in a new tab, but speaking of that...

3.  **"Ctrl-Enter" is a strange choice for a default keymapping to open Copilot in a new tab**  I know you can change this fairly easily, but the fact that the default keymapping is an incredibly common keystroke for a number of other operations (namely, running code in various languages and paradigms) seems to be somewhat of an oversight. 

This is an amazing tool, and I don't mean to be overly negative.  The folks that have created and worked on this are clearly a lot smarter and savvier than I can ever hope to be.  But, in my humble opinion, some of these issues are dragging down the ROI of Copilot.

Maybe I'm the only one having them?

**Comment by figloalds** [comment]
I want to add something but I don't think I should do a new discussion

On Visual Studio most of the times Co-Pilot decides to give a HUGE completely bogus suggestion instead of doing the 2 or so lines that I need. Here's an example of Co-Pilot filling my class with some bogus when all I needed was that it completed my set method

![image](https://user-images.githubusercontent.com/11756151/222439477-b19d4699-481b-46aa-8ab5-f25ffd7bb696.png)

Also it keeps breaking auto-completion when navigating between projects in a solution and yes sometimes it just doesn't work at all

It works significantly better on VS Code (Or maybe it works because the projects I work on VS Code are smaller)

**Reply by muralive** [reply]
you are not the only one, it does do some strange things at times
![image](https://github.com/community/community/assets/515146/2a31f3b5-1bc3-40d8-acb6-0cdbb8d65b4d)

**Comment by ZCG-coder** [comment]
<img width="604" alt="Screenshot 2024-05-11 at 1 55 36 PM" src="https://github.com/community/community/assets/76037593/a0b0aecc-7c20-4313-bed7-627bb2d17ad9">

It even says this...
